File: tests/fixtures/vulnerable/jwt_alg_none.ts
// Vulnerable: JWT verify without pinning algorithms
import jwt from "jsonwebtoken";

export function verifyToken(token: string) {
  // Bad: no algorithms option — attacker can choose
  return jwt.verify(token, "secret-key");
}

export function verifyTokenNoneAllowed(token: string) {
  // Bad: alg=none explicitly allowed
  return jwt.verify(token, "secret-key", { algorithms: ["HS256", "none"] });
}

File: tests/fixtures/vulnerable/jwt_alg_none.py
"""Vulnerable: JWT decode without pinned algorithms."""
import jwt


def verify_token(token):
    """Bad: no algorithms parameter — attacker can choose algorithm."""
    payload = jwt.decode(token, "secret-key")
    return payload


def verify_token_none_allowed(token):
    """Bad: alg=none explicitly allowed."""
    payload = jwt.decode(
        token, "secret-key", algorithms=["HS256", "none"]
    )
    return payload
